polynote-server/src/main/scala/polynote/server/NotebookSession.scala

```diff
@@ -165,7 +165,7 @@ class NotebookSession(
     subscriber.checkPermission(Permission.ReadNotebook).flatMap { _ =>
       // make sure to send status first, so client knows whether this notebook is up or not.
       sendStatus *> sendVersion *> sendNotebookContents *> sendTasks *> sendPresence *> sendCellStatuses
-    }.catchAll(err => PublishMessage(Error(0, err)))
```

Comment:
I don't really know why this wasn't working - but no error was ever being received by the UI under `msgType 0`, even when you purposely make the program fail.